### ISSUE TITLE
Bad error message with ChaiScript dynamic class object

### ISSUE THREAD

**Issue by lucas-sarie** [ChaiScript/ChaiScript#296]
Hi,

I have found a little bug with dynamic class method call, when a mistake is done on method name call the error doesn't correspond to the line where the call is done but it correspond to the function that call the erroneous function call (see exemple)

 * Compiler Used: MSVC13(x86)
 * Operating System: W7
 * Architecture (ARM/x86/32bit/64bit/etc): X64

### Expected Behavior 
raised exception like "Unknow MyFunc_ERR_NAME method" or something like this

### Actual Behavior 
raised eval_error with following content (pretty_print):
"
Error: "Error calling function 'f'" With parameters: () during evaluation at (C:\XXX\Script.txt 21,
)

  Expected: ()
          Defined at (C:\XXX\Script.txt 17, 6)

  C:\XXX\Script.txt (21, 1) 'f()'
"

### Minimal Example to Reproduce Behavior
```
class MyClass {
	attr x
	
	def MyClass()
	{
		this.x = 0.0
	}
	
	def MyFunc(double x)
	{
		this.x = x
	}
};

def f()
{
	var inst = MyClass()
	inst .MyFunc_WRONG_NAME(1.0)
}

f()
```

**Comment by lefticus** [comment]
I agree that the parser is giving you a weird error here, but it looks like there are two issues

The first is that `inst .MyFunc_WRONG_NAME(1.0)` has an extra space in it, which really isn't valid ChaiScript.

The other is that ChaiScript defined classes have a built in `missing_missing` function that will let you use dynamic properties of the object.

See https://github.com/ChaiScript/ChaiScript/blob/develop/cheatsheet.md#dynamic-objects for more details on the dynamic objects.

**Comment by lucas-sarie** [comment]
Sorry for the space it is a bad copy/paste from myself, but in my test case the space was not present,
About the method missing handler, if I had this at the begin of my script :
```
def method_missing(int i, string name, Vector v) {
  print("method_missing(${i}, ${name}), ${v.size()} params");
}
```
And the handler doesn't get called. (same error)
If I call `MyFunc_WRONG_NAME` outside `f` function I have a bad_cast exception but no message printed.
Finally if I call `5.bob(1,2,3);` inside or outside `f` function I got the good error message.

Thank you !

**Comment by lefticus** [comment]
I've fixed it here: 508729ec77949d801229a38f951c46985532bebc you now get a better error message when trying to all a non-function dynamic (method_missing) value on a Dynamic_Object
